insonmnia/dwh/sql.go

```diff
@@ -1116,6 +1085,51 @@ func (m *sqlStorage) CheckStaleID(conn queryConn, id *big.Int, entity string) (b
 	return true, nil
 }
 
+func (m *sqlStorage) InsertLastEvent(conn queryConn, event *blockchain.Event) error {
+	query, args, _ := m.builder().Insert("Misc").Columns("BlockNumber", "TxIndex", "Index").
+		Values(event.BlockNumber, event.TxIndex, event.Index).ToSql()
+	_, err := conn.Exec(query, args...)
+	return err
+}
+
+func (m *sqlStorage) UpdateLastEvent(conn queryConn, event *blockchain.Event) error {
+	query, args, _ := m.builder().Update("Misc").SetMap(map[string]interface{}{
+		"BlockNumber": event.BlockNumber,
+		"TxIndex":     event.TxIndex,
+		"Index":       event.Index,
+	}).ToSql()
+	_, err := conn.Exec(query, args...)
+	return err
+}
+
+func (m *sqlStorage) GetLastEvent(conn queryConn) (*blockchain.Event, error) {
+	query, _, _ := m.builder().Select("*").From("Misc").Limit(1).ToSql()
+	rows, err := conn.Query(query)
+	if err != nil {
+		return nil, fmt.Errorf("failed to GetLastEvent: %v", err)
+	}
+	defer rows.Close()
+
+	if ok := rows.Next(); !ok {
+		return nil, errors.New("GetLastEvent: no entries")
+	}
+
+	var (
```

Comment:
BTW why do you declare vars of the same type this way instead of var a, b, c uint64?